source/ipc-class.cpp

```diff
@@ -32,31 +32,18 @@ std::string ipc::collection::get_name() {
 }
 
 bool ipc::collection::register_function(std::shared_ptr<ipc::function> func) {
-	std::string fnId = func->get_unique_name();
+	std::string fnId = func->get_name();
 	if (m_functions.count(fnId) > 0)
 		return false;
 
 	m_functions.insert(std::make_pair(fnId, func));
 	return true;
 }
 
-std::shared_ptr<ipc::function> ipc::collection::get_function(const std::string& name, const std::vector<ipc::type>& params) {
-	std::string fnId = ipc::base::make_unique_id(name, params);
+std::shared_ptr<ipc::function> ipc::collection::get_function(const std::string& name) {
 	for (auto fct: m_functions) {
-		if (fct.first.compare(fnId.c_str()) == 0)
+		if (fct.first.compare(name.c_str()) == 0)
 			return fct.second;
 	}
-	// Not working on Mac
-	// if (m_functions.count(name) == 0)
-	// 	return nullptr;
-	return m_functions[fnId];
-}
-
-std::shared_ptr<ipc::function> ipc::collection::get_function(const std::string& name, const std::vector<ipc::value>& params) {
-	std::vector<ipc::type> argts;
-	argts.reserve(params.size());
-	for (const auto& v : params) {
-		argts.push_back(v.type);
-	}
-	return get_function(name, argts);
+	return m_functions[name];
```

Comment:
If we did not find a function name using loop above. 
What point of returning "m_functions[name];"

Reply:
true, making the change so we return a null ptr, good catch

Reply:
done.